aten/src/ATen/native/Pow.cpp

```diff
@@ -77,6 +77,70 @@ Tensor pow(Scalar base, const Tensor& exp) {
   return native::pow_out(result, base, exp);
 }
 
+Tensor& float_power_out(Tensor& result, const Tensor& base, const Tensor& exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ?
+                at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+
+  auto iter = TensorIterator::binary_op(result, base.to(dtype), exp.to(dtype));
+  pow_tensor_tensor_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor& float_power_out(Tensor& result, const Tensor& base, Scalar exp) {
+  // Numpy compatibility check:
+  TORCH_CHECK(!(isIntegralType(base.scalar_type(), true) &&
```

Comment:
I don't think NumPy implements this check for float_power:

```
a = np.array((1, -2, 3))
a.dtype
: dtype('int64')

np.float_power(a, -3)
: array([ 1.        , -0.125     ,  0.03703704])
```

Maybe I'm missing something?

Reply:
Yes, NumPy does not implements this compatibility check, removed!